runtime/near-vm-runner/src/wasmer2_runner.rs

```diff
@@ -40,25 +40,36 @@ const WASMER_FEATURES: wasmer::Features = wasmer::Features {
     signal_less: true,
 };
 
-pub struct Wasmer2Memory(Memory);
+#[derive(Clone)]
+pub struct Wasmer2Memory(Arc<LinearMemory>);
 
 impl Wasmer2Memory {
-    pub fn new(
-        store: &Store,
-        initial_memory_pages: u32,
-        max_memory_pages: u32,
-    ) -> Result<Self, VMError> {
-        Ok(Wasmer2Memory(
-            Memory::new(
-                store,
-                MemoryType::new(Pages(initial_memory_pages), Some(Pages(max_memory_pages)), false),
+    pub(crate) fn new(initial_memory_pages: u32, max_memory_pages: u32) -> Result<Self, VMError> {
+        let max_pages = Pages(max_memory_pages);
+        Ok(Wasmer2Memory(Arc::new(
+            LinearMemory::new(
+                &MemoryType::new(Pages(initial_memory_pages), Some(max_pages), false),
+                &MemoryStyle::Static { bound: max_pages, offset_guard_size: 0x1000 },
             )
-            .expect("TODO creating memory cannot fail"),
-        ))
+            .expect("creating memory must not fail"),
+        )))
+    }
+
+    fn view(&self) -> MemoryView<u8> {
```

Comment:
Seems like we can just `-> &[u8]` here?

Reply:
`MemoryView` derefs to `&[Cell<u8>]` really, allowing mutation as necessary for e.g. `write_memory`.

Reply:
Yeah, and it seems that this Deref is the only bit of MemoryView we use? So it seems that we can use `[Cell<u8>]` *directly* instead and avoid `MemoryView` abstraction altogether?

Reply:
Changed this area of code significantly (and added tests!)